lib/tools/adb-emu-commands.js

```diff
@@ -43,4 +65,45 @@ emuMethods.sendSMS = async function (phoneNumber, message = '') {
   await this.adbExecEmu(['sms', 'send', phoneNumber, message]);
 };
 
+emuMethods.gsmCall = async function (phoneNumber, action = '') {
+  if (GSM_CALL_ACTIONS.indexOf(action) === -1) {
+    log.errorAndThrow(`Invalid gsm action param ${action}`);
+  }
+  phoneNumber = `${phoneNumber}`.replace(/\s*/, "");
+  if (!PHONE_NUMBER_PATTERN.test(phoneNumber)) {
+    log.errorAndThrow(`Invalid phoneNumber param ${phoneNumber}`);
+  }
+  await this.adbExecEmu(['gsm', action, phoneNumber]);
+};
+
+emuMethods.gsmSignal = async function (strength = 4) {
+  if (!(strength in GSM_SIGNAL_STRENGTH)) {
+    log.errorAndThrow(`Invalid signal strength param ${strength}`);
+  }
+  log.info('gsm signal-profile <strength> changes the reported strength on next (15s) update.');
+  await this.adbExecEmu(['gsm', 'signal-profile', strength]);
+};
+
+emuMethods.gsmVoice = async function (state = 'on') {
+  // gsm voice <state> allows you to change the state of your GPRS connection
+  if (GSM_VOICE_STATES.indexOf(state) === -1) {
+    log.errorAndThrow(`Invalid gsm voice state param ${state}`);
+  }
+  await this.adbExecEmu(['gsm', 'voice', state]);
+};
+
+emuMethods.GSM_CALL = GSM_CALL;
```

Comment:
we might define all this like const emuMethods = {GSM_CALL: 'call' ...} and then use emuMethods.GSM_CALL... values in other constant declarations